k2/csrc/fsa_algo.cc

```diff
@@ -194,4 +193,111 @@ void ArcSort(const Fsa &a, Fsa *b,
   }
   if (arc_map != nullptr) arc_map->swap(indexes);
 }
+
+void TopSort(const Fsa &a, Fsa *b,
```

Comment:
I think make boolean as the return value may be better, as the caller will know what it should do next according to the return value:

- if the input fsa is empty, we return true, as sorted fsa on empty fsa is still empty.
- if the input fsa is non-connected or cyclic, we return false. Then the caller knows that the input is invalid, the caller may check the input fsa.

For now, as we always clear the output fsa at the beginning, the caller can not tell the difference when the function returns from different branches.

Reply:
Fixed.